.github/workflows/kubernetes-create.yml

```
@@ -0,0 +1,174 @@
+name: Kubernetes Manifest Creator Reusable
+
+on:
+  workflow_call:
+    inputs:
+      certificate:
+        type: string
+        description: ARN certificate for the domain
+        required: true
+        default: ''
+      cluster:
+        type: string
+        description: Loadbalancer cluster name
+        default: '23g-a'
+        required: false
+      hostname:
+        type: string
+        description: Hostname
+        required: true
+        default: ''
+      project:
+        type: string
+        description: Projectname (lowercase, only a-z,0-9 or - and must start with a-z!)
+        required: true
+        default: ''
+    secrets:
+      shared-iam-role:
+        required: true
+      shared-s3-bucket:
+        required: true
+
+jobs:
+  deploy:
+    name: Kubernetes Manifest Creator
+    runs-on: ubuntu-latest
+    env:
+      certificate: ${{ inputs.certificate }}
+      cluster: ${{ inputs.cluster }}
+      hostname: ${{ inputs.hostname }}
+      project: ${{ inputs.project }}
+    permissions: write-all
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Check for existing Manifest
+        id: checker
+        run: |
+          if test -f "kubernetes/${{ env.project }}.yaml"; then
+            echo "Manifest already exists! Please remove/rename kubernetes/${{ env.project }}.yaml and retry."
+            exit 1
+          fi
+
+      - name: Configure AWS credentials for account
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.shared-iam-role }}
+          aws-region: eu-west-1
+
+      - name: Check/Create Parameter Store
+        id: create-parameter-store
+        run: |
+          aws ssm put-parameter --name ${{ env.project }} --region eu-west-1 --value "ENV" --type SecureString
+
+      - name: Check/Create ECR Repository
+        id: create-ecr
+        run: |
+          aws ecr describe-repositories --region eu-west-1 --repository-names ${{ env.project }} || aws ecr create-repository --region eu-west-1 --repository-name ${{ env.project }}
+
+      - name: Create S3 bucket
+        id: CustomerS3
+        run: |
+          aws s3api head-bucket --bucket 23g-sharedhosting-${{ env.project }} --region eu-west-1 || aws s3api create-bucket --bucket 23g-sharedhosting-${{ env.project }} --region eu-west-1 --create-bucket-configuration LocationConstraint=eu-west-1 --acl public-read
```

Heb zelf dit gedaan bij Recirculo.
```suggestion
          aws s3api head-bucket --bucket 23g-sharedhosting-${{ env.project }} --region eu-west-1 || aws s3api create-bucket --bucket 23g-sharedhosting-${{ env.project }} --region eu-west-1 --create-bucket-configuration LocationConstraint=eu-west-1
```

---

Hm, ja dit moet eigenlijk wel configureerbaar zijn IMO